packages/hydrogen/edge-entry.js

```diff
@@ -3,7 +3,15 @@ import indexTemplate from '__RELATIVE__/dist/client/index.html?raw';
 
 // ReadableStream is bugged in Vercel Edge, overwrite with polyfill
 import { ReadableStream } from 'web-streams-polyfill/ponyfill';
-Object.assign(globalThis, { ReadableStream });
+Object.assign(globalThis, {
+  // ReadableStream is bugged in Vercel Edge, overwrite with polyfill
+  ReadableStream,
+
+  // Hydrogen exposes env vars through `Oxygen.env`
+  Oxygen: {
+    env: process.env
```

Comment:
Unfortunately I actually think that this won't work the way you're expecting. This is because for Edge functions, `process.env` usage is statically analyzed to determine which variables to include in the edge function bundle.

To make it work with `Oxygen.env` usage then we would have to do similar static analysis in `@vercel/hydrogen` and populate the `envVarsInUse` array in the `EdgeFunction` constructor. Perhaps Hydrogen could provide that information if it is already doing static analysis during the build process?

Reply:
We're not doing any static analysis in the build process in Hydrogen, afaik 🤔 
Perhaps you can replace `Oxygen.env.XYZ` matches in `@vercel/hydrogen`?

Also, [we are moving to using `Hydrogen.env` instead of `Oxygen.env`](https://github.com/Shopify/hydrogen/pull/2150) to avoid confusion when deploying to other platforms, so something to keep in mind.